buffer.go

```diff
@@ -0,0 +1,159 @@
+package stats
+
+import (
+	"io"
+	"sync"
+	"sync/atomic"
+	"time"
+	"unsafe"
+)
+
+// Buffer is the implementation of a measure handler which uses a Serializer to
+// serialize the metric into a memory buffer and write them once the buffer has
+// reached a target size.
+type Buffer struct {
+	// Target size of the memory buffer where metrics are serialized.
+	//
+	// If left to zero, a size of 1024 bytes is used as default (this is low,
+	// you should set this value).
+	BufferSize int
+
+	// The Serializer used to write the measures.
+	//
+	// This field cannot be nil.
+	Serializer Serializer
+
+	buffer unsafe.Pointer
```

Comment:
It has to be an unsafe pointer so we can perform an atomic compare-and-swap on the value.
I wish we could use `atomic.Value` to get a bit more type safety but it only supports atomic load and store.